ai/nl_agent.py

```
@@ -0,0 +1,284 @@
+        },
+        "pin": {
+            "beginner":     "A pin stops a piece from moving because a more valuable piece is behind it.",
+            "intermediate": "A pin restricts a piece that shields a more valuable piece from capture.",
+            "advanced":     "Absolute pins (king behind) are legally binding; relative pins create strategic pressure.",
+        },
+        "skewer": {
+            "beginner":     "A skewer forces a valuable piece to move, exposing a weaker piece behind it.",
+            "intermediate": "A skewer attacks the more valuable piece first, winning the piece behind it.",
+            "advanced":     "Skewers invert the pin dynamic and are most powerful when the rear piece is undefended.",
+        },
+    }
```

Ruff (0.15.11)

[warning] Mutable default value for class attribute

(RUF012)